pkg/downloader/chart_downloader.go

```diff
@@ -378,19 +377,13 @@ func (c *ChartDownloader) scanReposForURL(u string, rf *repo.File) (*repo.Entry,
 		}
 
 		idxFile := filepath.Join(c.RepositoryCache, helmpath.CacheIndexFile(r.Config.Name))
-		i, err := repo.LoadIndexFile(idxFile)
+		yamlFile, err := os.ReadFile(idxFile)
 		if err != nil {
 			return nil, errors.Wrap(err, "no cached repo found. (try 'helm repo update')")
 		}
-
-		for _, entry := range i.Entries {
-			for _, ver := range entry {
-				for _, dl := range ver.URLs {
-					if urlutil.Equal(u, dl) {
-						return rc, nil
-					}
-				}
-			}
+		file := string(yamlFile[:])
+		if strings.Contains(file, u) {
+			return rc, nil
```

Comment:
I'm not sure about this change. The `string.Contains` is run over a file with annotations and descriptions. The `u` could be in one of those fields which would return an improper result. While this is unlikely, the code that calls `scanReposForURL` grabs the credentials if there are any to pass on. I wonder if there is an edge case that could leak credentials with misuse of this change.

Reply:
I have that reverted now, so hopefully it can be merged now! If anything else needs changing let me know